docs/testing-guidelines/getting-code-coverage.md

```diff
@@ -0,0 +1,127 @@
+Getting Code Coverage Analysis for PowerShell
+===
+
+**Note: Code coverage is currently only supported on Windows, since we use OpenCover**
+
+The PowerShell code base is configured to build with code coverage support using [OpenCover](https://github.com/OpenCover/opencover). You can see the testing coverage of the current [`master`](https://github.com/PowerShell/PowerShell) branch build at any time at [codecov.io](https://codecov.io/gh/PowerShell/PowerShell).
+
+To run test coverage analysis of PowerShell on your own branch/machine
+however, you will need to take the following steps (and be aware that running
+the tests under code coverage analysis can take as long as 8 hours).
+
+Running tests with code coverage analysis
+---
+
+**First**: Open PowerShell in an **Administrator** session. OpenCover
+needs Administrator privileges to work.
+
+Now, in PowerShell:
+```powershell
+# Go to your PowerShell build directory root
+PS> Set-Location "C:\Path\to\powershell\build\dir"
+
+# Import the PowerShell build module
+PS> Import-Module .\build.psm1
+
+# Build PowerShell. You may need to add other flags here like
+# -ResGen, -Restore or -PSModuleRestore
+PS> Start-PSBuild -Configuration CodeCoverage
+
+# Save the built executable's location for later use
+PS> $psDir = Split-Path -Parent (Get-PSOutput)
+
+# Now ensure Pester is installed
+PS> Restore-PSPester
+
+# Import the OpenCover module
+PS> Import-Module $PWD\test\tools\OpenCover
+
+# Install OpenCover to a temporary directory
+PS> Install-OpenCover -TargetDirectory $env:TEMP -Force
+
+# Finally, run the tests with code coverage analysis.
+# If you want to run only the continuous integration tests,
+# add -CIOnly, which will take less time
+PS> Invoke-OpenCover `
+>> -OutputLog coverage.xml `
```

Comment:
I believe we also have reasonable defaults for these

Reply:
If we use the default `OpenCoverPath` in `Install-OpenCover`, we should do that here too.

I assume the function is designed to be run in a specific place, and in that case has sane defaults for `TestPath` and `PowerShellExeDirectory` (in which case we can take out the line defining `$psDir`).

But in the case of `OutputLog` I think we should keep that so anyone following along knows where the output will be.

In general, I'm sort of in favour of using explicit parameters in the documentation, only because there's more information to work with if something goes wrong. It's why PowerShell has named parameters after all.

Reply:
I say all this after having to read the scripts to work out what the parameters do

Reply:
The defaults here are:
* `OutputLog`: `"$home/Documents/OpenCover.xml"`
* `TestPath`: `"${script:psRepoPath}/test/powershell"`
* `OpenCoverPath`: `"$home/OpenCover"`
* `PowerShellExePath`: `"${script:psRepoPath}/src/powershell-win-core/bin/CodeCoverage/netcoreapp2.0/win7-x64/publish"`

So I feel like `TestPath` and `PowerShellExePath` are pretty sane and we can omit them happily.

For `OpenCoverPath` we should only use the default path if `Install-OpenCover` does (I'm not a huge fan of installing test tools in `$HOME`, especially if the user has no knowledge of that, but we could change the default).

And `OpenCoverPath` I think is definitely better being explicit. Nobody will ever find `OpenCover.xml` in their `$HOME` directory.